### ISSUE TITLE
Registering custom dataset

### ISSUE THREAD

**Issue by Martin9797** [shanice-l/gdrnpp_bop2022#49]
I am trying to train on my own custom dataset, but there seems to be an issue when trying to register the dataset for detectron2. I have already written files in the det/yolox/data/datasets folder, the /ref folder, added it to the dataset_factory.py file and wrote a config file for the dataset. 
The error I get looks like this:
```
[0329_144020 det.yolox.engine.yolox_setup@121]: Full config saved to /home/faps/GDRNPP_pose_est/pose_est_gdrnpp/output/yolox/bop_pbr/yolox_x_640_augCozyAAEhsv_ranger_30_epochs_Stator_test/config.yaml
[0329_144020 det.yolox.engine.yolox_setup@125]: seed: -1
[0329_144020 d2.utils.env@41]: Using a generated random seed 21122847
[0329_144020 det.yolox.engine.yolox_setup@146]: Used mmcv backend: cv2
[0329_204020@Stator_test:411] DBG register dataset: Stator_train
ERROR [0329_144020 d2.engine.launch@84]: An error has been caught in function 'launch', process 'MainProcess' (430301), thread 'MainThread' (139953541154048):
Traceback (most recent call last):

  File "./det/yolox/tools/main_yolox.py", line 77, in <module>
    args=(args,),
          -> Namespace(config_file='configs/yolox/bop_pbr/yolox_x_640_augCozyAAEhsv_ranger_30_epochs_Stator_test.py', dist_url='tcp://127....

> File "/home/faps/anaconda3/envs/GDRNPP/lib/python3.7/site-packages/detectron2/engine/launch.py", line 84, in launch
    main_func(*args)
    |          -> (Namespace(config_file='configs/yolox/bop_pbr/yolox_x_640_augCozyAAEhsv_ranger_30_epochs_Stator_test.py', dist_url='tcp://127...
    -> <function main at 0x7f4852409440>

  File "./det/yolox/tools/main_yolox.py", line 50, in main
    cfg = setup(args)
          |     -> Namespace(config_file='configs/yolox/bop_pbr/yolox_x_640_augCozyAAEhsv_ranger_30_epochs_Stator_test.py', dist_url='tcp://127....
          -> <function setup at 0x7f49782177a0>

  File "./det/yolox/tools/main_yolox.py", line 43, in setup
    register_datasets_in_cfg(cfg)
    |                        -> {'train': {'output_dir': '/home/faps/GDRNPP_pose_est/pose_est_gdrnpp/output/yolox/bop_pbr/yolox_x_640_augCozyAAEhsv_ranger_30...
    -> <function register_datasets_in_cfg at 0x7f4852400680>

  File "/home/faps/GDRNPP_pose_est/pose_est_gdrnpp/det/yolox/tools/../../../det/yolox/data/datasets/dataset_factory.py", line 106, in register_datasets_in_cfg
    assert "DATA_CFG" in cfg and name in cfg.DATA_CFG, "no cfg.DATA_CFG.{}".format(name)
                         |       |       |                                         -> 'Stator_test'
                         |       |       -> {'train': {'output_dir': '/home/faps/GDRNPP_pose_est/pose_est_gdrnpp/output/yolox/bop_pbr/yolox_x_640_augCozyAAEhsv_ranger_30...
                         |       -> 'Stator_test'
                         -> {'train': {'output_dir': '/home/faps/GDRNPP_pose_est/pose_est_gdrnpp/output/yolox/bop_pbr/yolox_x_640_augCozyAAEhsv_ranger_30...

AssertionError: no cfg.DATA_CFG.Stator_test
```

Is there a step I am missing to register the dataset? 

Thank you for your help!

**Comment by Yuan-TS** [comment]
Hello, I also encountered the same problem, how did you solve it? @Martin9797

**Comment by cnalty** [comment]
@Yuan-TS If you haven't figured it out already, I have gotten past this error by doing a few things. 

1. Add a file to the ref folder for your dataset, I just called mine custom.py. I copied over one of the other datasets ref files and replaced the info with my datasets info. also add this file to __init__.py in ref
2. In core/gdrn_modeling/datasets I copied one of the dataset files here, and again switched out all the info with my datasets info
3. in core/gdrn_modelling/datasets at the top of dataset_factor.py import my new custom dataset file created in 2, and add the name to _DSET_MOD_NAMES
